herddb-core/src/main/java/herddb/core/TableManager.java

```diff
@@ -2054,20 +2063,33 @@ private TableCheckpoint checkpoint(double dirtyThreshold, double fillThreshold,
 
             pageAnalysis = System.currentTimeMillis();
 
+            /* Should currently new rebuild page kept on memory or discarded? */
+            boolean keepFlushedPageInMemory = false;
+
             /* Rebuild dirty pages with only records to be kept */
             for (WeightedPage weighted : flushingDirtyPages) {
 
                 /* Page flushed */
                 flushedPages.add(weighted.pageId);
                 ++flushedDirtyPages;
 
-                final DataPage dataPage = pages.get(weighted.pageId);
+                final boolean currentPageWasInMemory;
+                final DataPage dataPage = pages.remove(weighted.pageId);
```

Comment:
is this 'remove' needed ?

I am thinking about the case in which the checkpoint is running and we have a query (currently this is not possible but in the future it will be), if you remove the pages from 'pages' the page won't be more usable.
Shouldn't we remove the page only after appending the new pages ?